src/scraping/css.py

```diff
@@ -0,0 +1,263 @@
+# Copyright 2020 CDPedistas (see AUTHORS.txt)
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranties of
+# MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For further info, check  https://github.com/PyAr/CDPedia/
+
+"""CSS stylesheets scraping."""
+
+import functools
+import logging
+import os
+import re
+import threading
+import urllib.error
+import urllib.parse
+import urllib.request
+
+import config
+from src import utiles
+from src.scraping.pydocs import HEADERS
+
+logger = logging.getLogger(__name__)
+
+# regex matching links from the 'url' CSS function
+re_resource_url = re.compile(r'url\(([^\s}]+)\)')
+
+
+class _LinkExtractor:
+    """Collect CSS stylesheet links from the HTML source code."""
+
+    def __init__(self):
+        self.cssdir = None
+        self.links = None
+
+        # pattern for extracting css links from html
+        regex = r"/w/load.php\?.*?only=styles&amp;skin=vector"
+        self._findlinks = re.compile(regex).findall
+
+        # lock for writing to same file from different threads
+        self._lock = threading.Lock()
+
+    def setup(self, cssdir):
+        """Set output file handler and load previously saved data."""
+        self.cssdir = cssdir
+        links_file = os.path.join(cssdir, config.CSS_LINKS_FILENAME)
+        try:
+            self._fh = open(links_file, 'r+t', encoding='utf-8', buffering=1)
+            self.links = set(line.strip() for line in self._fh)
```

Comment:
Acá podés hacer directamente un set comprehension: `{line.strip() for line in self._fh}`